apollo-federation/src/composition/mod.rs

```diff
@@ -0,0 +1,57 @@
+mod satisfiability;
+
+use std::vec;
+
+use crate::error::FederationError;
+use crate::schema::schema_upgrader::upgrade_subgraphs_if_necessary;
+use crate::subgraph::typestate::Expanded;
+use crate::subgraph::typestate::Initial;
+use crate::subgraph::typestate::Subgraph;
+use crate::subgraph::typestate::Upgraded;
+use crate::subgraph::typestate::Validated;
+use crate::supergraph::Satisfiable;
+use crate::supergraph::Supergraph;
+
+pub fn compose(
+    subgraphs: Vec<Subgraph<Initial>>,
+) -> Result<Supergraph<Satisfiable>, Vec<FederationError>> {
+    let expanded_subgraphs = expand_subgraphs(subgraphs)?;
+    let upgraded_subgraphs = upgrade_subgraphs_if_necessary(expanded_subgraphs)?;
+    let _validated_subgraphs = validate_subgraphs(upgraded_subgraphs)?;
+
+    todo!("implement magic here")
+    // let supergraph = merge_subgraphs(validated_subgraphs)?;
+    // validate_satisfiability(supergraph)?
```

Comment:
Don't `merge_subgraphs` and `validate_satisfiability` both panic already? If so, can we remove this `todo!` and just let those functions panic (and remove the `allow_unused`)?

Reply:
indeed there is an existing `merge_subgraphs` function that was created 1.5 years ago (that connectors was relying on)... but that needs to be updated to use the typestate stuff and didn't want to touch it yet (as it is a live code)

Reply:
I'll create new stub for `merge_subgraphs`

Reply:
4ca6564e535b7390a006615389662d66e59648aa